### PR TITLE
1.0 adapter support for mantis

### PR DESCRIPTION
**Pull request by parisholley**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

### CODE REVIEW

File: modules/mantisBidAdapter.js [hunk 1]
```diff
@@ -1,228 +1,221 @@
...
-        queryKey = chain + '[' + key + ']';
-      }
+function isAmp() {
+  return typeof window.context === 'object' && (window.context.tagName === 'AMP-AD' || window.context.tagName === 'AMP-EMBED');
+}
 
-      var val = data[key];
+function isSecure() {
+  return document.location.protocol === 'https:';
+}
```
Comment (idettman) [inline modules/mantisBidAdapter.js]: Note: There is a function for testing types included in [utils.js](https://github.com/prebid/Prebid.js/blob/master/src/utils.js) beginning on line #332. Line #334 includes an isArray method where you can see how the "isA" method could be used for testing types not included such as "[Object]".

**Comment by mkendall07** [comment]
@idettman what's the status on this one?

**Comment by mkendall07** [comment]
@parisholley 
Please resolve the conflicts.

**Comment by mkendall07** [comment]
@parisholley 
needs rebasing before this can be merged.

**Comment by parisholley** [comment]
@mkendall07 done

File: modules/mantisBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  }
+  window.mantis_uuid = uuid;
+  if (window.localStorage) {
+    try {
+      window.localStorage.setItem('mantis:uuid', uuid);
+    } catch (ex) {
+    }
+  }
+  if (!sync) {
+    postMessage('uuid', uuid);
```
Comment (mkendall07) [inline modules/mantisBidAdapter.js]: what's the purpose of the postMessage?

File: modules/mantisBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      url: buildMantisUrl('/prebid/display', query) + '&foo',
+      data: ''
+    };
+  },
+  interpretResponse: function (serverResponse) {
+    storeUuid(serverResponse.uuid);
+    return serverResponse.body.ads.map(function (ad) {
+      return {
+        requestId: ad.bid,
+        bidderCode: spec.code,
```
Comment (mkendall07) [inline modules/mantisBidAdapter.js]: Please drop this line. We add the code automatically from the spec, and it breaks aliasing if this is here.

Comment (parisholley) [reply modules/mantisBidAdapter.js]: @mkendall07 done

File: modules/mantisBidAdapter.js (hunk 2 above)
Comment (parisholley) [reply modules/mantisBidAdapter.js]: @mkendall07 this was common code for frame syncing for our non-prebid code, build now removes it. thanks

**Comment by mkendall07** [comment]
hey @parisholley 
LGTM. Just needs a corresponding docs PR to merge: Please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html).

**Comment by parisholley** [comment]
@mkendall07 done, thanks!

https://github.com/prebid/prebid.github.io/pull/546

**Comment by rmloveland** [comment]
Added the needs-docs label since it's not showing up on http://prebid.org/download.html in the 1.2.0 dropdown due to lack of the page having the "prebid 1.0 support" key in the page's YAML header.

**Pull request merged**